app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -225,10 +225,10 @@ class HomeFragment : Fragment() {
         activity.themeManager.applyStatusBarTheme(activity)
 
         view.consumeFrom(requireComponents.core.store, viewLifecycleOwner) {
-            val tabCount = if (currentMode.getCurrentMode() == Mode.Normal) {
-                it.normalTabs.size
-            } else {
+            val tabCount = if (browsingModeManager.mode.isPrivate) {
```

Comment:
An exhaustive `when` would have prevented this bug, but i guess we can't do that w/ sealed classes?

Reply:
Actually, you can switch to an exhaustive `when` here, so let's do that instead.

Reply:
We were just using the wrong way to get the mode. `getCurrentMode()` returns `Mode` -> private, normal, onboarding https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/home/Mode.kt#L21-L24

Now this is just a boolean (which is how we use it everywhere else that doesn't care about onboarding status): https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/browser/browsingmode/BrowsingModeManager.kt#L18

Reply:
Ah, cool -  i've misread that `if` statemen. That's what i get for writing code, reviewing code and listening to a meeting all at once.